app/lib/shared/exceptions.dart

```diff
@@ -157,14 +158,28 @@ class InvalidInputException extends ResponseException {
     _check(_ulidPattern.hasMatch(value), () => '"$name" is not a valid ulid.');
   }
 
-  static void checkSemanticVersion(String? version) {
+  /// Throw [InvalidInputException] if [package] is not an allowed package name.
+  static void checkPackageName(String? package) {
+    // TODO: Reuse logic from validatePackageName in pub_package_reader.dart
+    //       This is for new existing packages! Not only new packages!
```

Comment:
`new existing` should be only `existing`?

Reply:
I think it's only supposed to check the format of the name, so both new and existing.

Reply:
`new AND existing` then?